### ISSUE TITLE
Installing with PHP7 fails due to doctrine/instantiator

### ISSUE THREAD

**Issue by bencagri** [doctrine/mongodb-odm#1648]
Hi, 
When trying to install mongodb-odb, due to instantiator, it fails. 

```
"doctrine/instantiator": "~1.0.1",
```
This version supports php: ~5.3
On the other hand v.1.1.0 supports php: ^7.1

How can I achieve this?

```
composer require doctrine/mongodb-odm-bundle
Using version ^3.4 for doctrine/mongodb-odm-bundle
./composer.json has been updated
Loading composer repositories with package information
Updating dependencies (including require-dev)
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - doctrine/instantiator 1.0.3 requires php ~5.3 -> your PHP version (7.1.9) does not satisfy that requirement.
    - doctrine/instantiator 1.0.2 requires php ~5.3 -> your PHP version (7.1.9) does not satisfy that requirement.
    - doctrine/instantiator 1.0.1 requires php ~5.3 -> your PHP version (7.1.9) does not satisfy that requirement.
    - Installation request for doctrine/mongodb-odm-bundle ^3.4 -> satisfiable by doctrine/mongodb-odm-bundle[3.4.0].
```

**Comment by alcaeus** [comment]
There's something else going on - `doctrine/instantiator` 1.0.5 supports PHP 7 as well. Could you please post the output of `composer why-not doctrine/instantiator 1.0.5`?

**Comment by bencagri** [comment]
Hi @alcaeus 

```
composer why-not doctrine/instantiator 1.0.5
There is no installed package depending on "doctrine/instantiator" in versions not matching 1.0.5
```

The problem is here, I think;
https://github.com/doctrine/mongodb-odm/blob/eb1ffc362f8271513592990b74dfcd4d126411e6/composer.json#L24

**Comment by alcaeus** [comment]
I'll create a pull request to relax it - however, `~1.0.1` should also allow `1.0.5`, that's why I'm wondering.

**Comment by bencagri** [comment]
I suppose it should be `^1.0`

**Comment by alcaeus** [comment]
Yes. You can create a pull request targeting the `1.1.x` branch if you'd like, but as I said, `1.0.5` will be eligible for installation in your case. I'll play around with this later today and figure out why it's not behaving as I'd expect it to.

**Comment by alcaeus** [comment]
This is strange:
```
$ php -v
PHP 7.0.23 (cli) (built: Sep 25 2017 08:07:30) ( NTS )
Copyright (c) 1997-2017 The PHP Group
Zend Engine v3.0.0, Copyright (c) 1998-2017 Zend Technologies

$ composer require doctrine/mongodb-odm-bundle
Using version ^3.4 for doctrine/mongodb-odm-bundle
./composer.json has been updated
Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 32 installs, 0 updates, 0 removals
  - Installing symfony/filesystem (v3.3.9): Loading from cache
  - Installing symfony/config (v3.3.9): Loading from cache
  - Installing psr/log (1.0.2): Loading from cache
  - Installing symfony/debug (v3.3.9): Loading from cache
  - Installing symfony/polyfill-mbstring (v1.5.0): Loading from cache
  - Installing symfony/console (v3.3.9): Loading from cache
  - Installing psr/container (1.0.0): Loading from cache
  - Installing symfony/dependency-injection (v3.3.9): Loading from cache
  - Installing symfony/stopwatch (v3.3.9): Loading from cache
  - Installing symfony/routing (v3.3.9): Loading from cache
  - Installing symfony/http-foundation (v3.3.9): Loading from cache
  - Installing symfony/event-dispatcher (v3.3.9): Loading from cache
  - Installing symfony/http-kernel (v3.3.9): Loading from cache
  - Installing symfony/finder (v3.3.9): Loading from cache
  - Installing symfony/class-loader (v3.3.9): Loading from cache
  - Installing symfony/polyfill-apcu (v1.5.0): Loading from cache
  - Installing psr/simple-cache (1.0.0): Loading from cache
  - Installing psr/cache (1.0.1): Loading from cache
  - Installing symfony/cache (v3.3.9): Loading from cache
  - Installing doctrine/cache (v1.6.2): Loading from cache
  - Installing symfony/framework-bundle (v3.3.9): Loading from cache
  - Installing doctrine/lexer (v1.0.1): Loading from cache
  - Installing doctrine/annotations (v1.4.0): Loading from cache
  - Installing doctrine/inflector (v1.2.0): Loading from cache
  - Installing doctrine/collections (v1.4.0): Loading from cache
  - Installing doctrine/common (v2.7.3): Loading from cache
  - Installing symfony/doctrine-bridge (v3.3.9): Loading from cache
  - Installing symfony/options-resolver (v3.3.9): Loading from cache
  - Installing doctrine/mongodb (1.6.0): Loading from cache
  - Installing doctrine/instantiator (1.0.5): Loading from cache
  - Installing doctrine/mongodb-odm (1.1.6): Loading from cache
  - Installing doctrine/mongodb-odm-bundle (3.4.0): Loading from cache
```

As you can see, `doctrine/instantiator` is correctly installed in the 1.0.5 version, so I'd assume there's something else that's going on. You don't have ORM installed by any chance?

**Comment by bencagri** [comment]
```
"require": {
        "php": ">=7.0",
        "alcaeus/mongo-php-adapter": "^1.1",
        "api-platform/core": "^2.1",
        "doctrine/doctrine-bundle": "^1.6",
        "doctrine/orm": "^2.5",
        "friendsofsymfony/user-bundle": "~2.0",
        "guzzlehttp/guzzle": "^6.0",
        "incenteev/composer-parameter-handler": "^2.0",
        "nelmio/cors-bundle": "^1.4",
        "phpdocumentor/reflection-docblock": "^3.0",
        "sensio/distribution-bundle": "^5.0.19",
        "sensio/framework-extra-bundle": "^3.0.2",
        "symfony/monolog-bundle": "^3.1.0",
        "symfony/swiftmailer-bundle": "^2.3.10",
        "symfony/symfony": "3.3.*",
        "twig/twig": "^1.0 || ^2.0"
    },
```

doctrine/orm is intalled.

**Comment by alcaeus** [comment]
Thanks for providing a `composer.json`, I was able to run a few more checks. Turns out, 5.3 is the last version that doesn't run on PHP 7. Could it be that you previously had 1.0.3 installed and never updated to 1.0.5? That would explain your problem, since I was able to install with the dependencies mentioned above.

Running `composer update doctrine/instantiator` followed by the `require` command you were originally trying to run should hopefully fix your issue.
